Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp

```diff
@@ -529,6 +529,11 @@ void WebLoaderStrategy::scheduleLoadFromNetworkProcess(ResourceLoader& resourceL
     if (loadParameters.isMainFrameNavigation)
         existingNetworkResourceLoadIdentifierToResume = std::exchange(m_existingNetworkResourceLoadIdentifierToResume, std::nullopt);
     WEBLOADERSTRATEGY_RELEASE_LOG("scheduleLoad: Resource is being scheduled with the NetworkProcess (priority=%d, existingNetworkResourceLoadIdentifierToResume=%" PRIu64 ")", static_cast<int>(resourceLoader.request().priority()), valueOrDefault(existingNetworkResourceLoadIdentifierToResume).toUInt64());
+
+    auto sendResult = WebProcess::singleton().ensureNetworkProcessConnection().connection().sendSync(Messages::NetworkConnectionToWebProcess::AllowsFirstPartyForCookies(RegistrableDomain { loadParameters.request.firstPartyForCookies() }), 0);
```

Comment:
Absolutely not.  Introducing a synchronous message send here would kill performance and cause hangs whenever anything is loaded.

Reply:
Ok. I'll change this to keep track of the allowedFirstPartyForCookies in the web process so we don't need this synchronous IPC call.